not written: save photos in database, some functions of volunteer

EkaterinaShulga opened the pull request.

EkaterinaShulga committed:
not written: save photos in database, some functions of volunteer

dmitry-bizin commented on pom.xml:
```diff
@@ -1,54 +1,114 @@
+
+        <dependency>
+            <groupId>org.springdoc</groupId>
+            <artifactId>springdoc-openapi-ui</artifactId>
```
вот у тебя выше подключена библиотека springfox - это другая реализация openapi по сути, поэтому не нужно их сразу обе использовать - либо springfox, либо springdoc надо

dmitry-bizin commented on pom.xml:
```diff
@@ -1,54 +1,114 @@
+        </dependency>
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>postgresql</artifactId>
```
оо, классно 👍

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/controller/ContactController.java:
```diff
@@ -0,0 +1,85 @@
+@RequestMapping("/contact")
+public class ContactController {
+
+    private final ContactServiceImpl contactServiceImpl;
```
ContactService - нужно инжектить интерфейс, а не реализацию, чтобы не привязываться к реализации

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/entity/Contact.java:
```diff
@@ -0,0 +1,62 @@
+
+    @OneToOne()
+    @JoinColumn(name = "id_client")
+    Client client;
```
private

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/CatButtonAnswers.java:
```diff
@@ -0,0 +1,146 @@
+        log.info(callBackData);
+        long chatId = update.callbackQuery().message().chat().id();
+        switch (callBackData) {
+            case "Узнать информацию о приюте":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/CatButtonAnswers.java:
```diff
@@ -0,0 +1,146 @@
+        int messageId = update.callbackQuery().message().messageId();
+        log.info("Ответ от кнопки " + answerMenu);
+        switch (answerMenu) {
+            case "infoCat":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/DogButtonAnswers.java:
```diff
@@ -0,0 +1,155 @@
+        log.info(callBackData);
+        long chatId = update.callbackQuery().message().chat().id();
+        switch (callBackData) {
+            case "Узнать информацию о приюте":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/DogButtonAnswers.java:
```diff
@@ -0,0 +1,155 @@
+        // List<Shelter> shelters = shelterRepository.findAll();
+        // List<InformationForOwner> info = informationForOwnerRepository.findAll();
+        switch (answerMenu) {
+            case "info":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/StatusChecker.java:
```diff
@@ -0,0 +1,168 @@
+        int messageId = update.callbackQuery().message().messageId();
+        long chatId = update.callbackQuery().message().chat().id();
+        switch (callBackData) {
+            case "Волонтер":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/StatusChecker.java:
```diff
@@ -0,0 +1,168 @@
+        client.setChatId(chatId);
+        client.setStatus(0);
+        switch (callBackData) {
+            case "Приют для собак":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/StatusChecker.java:
```diff
@@ -0,0 +1,168 @@
+        volunteer.setStatus(0);
+
+        switch (callBackData) {
+            case "Волонтер приюта для кошек":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/VolunteerButtonAnswers.java:
```diff
@@ -0,0 +1,48 @@
+        long chatId = update.callbackQuery().message().chat().id();
+        log.info("Ответ от кнопки " + callBackData);
+        switch (callBackData) {
+            case " Добавить усыновителя ":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/handler/button_answers/VolunteerButtonAnswers.java:
```diff
@@ -0,0 +1,48 @@
+                log.warn("send information for volunteer - how create new user");
+                telegramBot.execute(message);
+                break;
+            case " Добавить питомца ":
```
вот строки лучше бы вынести в константы, чтобы самой не запутаться, где какую отправлять и проверять и ошибки так избежишь, что какая-то кнопка не работает

dmitry-bizin commented on src/main/java/ru/skypro/shelterforanimals/repository/RecordRepository.java:
```diff
@@ -0,0 +1,21 @@
+public interface RecordRepository extends JpaRepository<Record, Long> {
+
+    Record findByRecordId(long recordId);
+    LinkedList<Record> findAllRecordByChatId(long chatId);
```
List

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -1 +1,13 @@
+spring.datasource.username=cynologist
+spring.datasource.password=12345
 
+spring.jpa.hibernate.ddl-auto=update
```
validate

EkaterinaShulga committed:
весь основной функционал бота реализован
комментарии к методам написаны
удален лишний код и комментарии
код выровнен
доделать: тесты, ньюансы по функционалу

EkaterinaShulga committed:
Delete ReportService.java

EkaterinaShulga committed:
Delete ReportServiceImpl.java

EkaterinaShulga committed:
Delete Report.java
